design-proposals/flavors-and-preferences.md

```diff
@@ -0,0 +1,1023 @@
+	// +optional
+	Realtime *v1.Realtime `json:"realtime,omitempty"`
+}
+```
+
+### `MemoryFlavor`
+
+```go
+// FlavorMemory
+//
+// +k8s:openapi-gen=true
+type MemoryFlavor struct {
+
+	// Guest allows to specifying the amount of memory which is visible inside the Guest OS.
+	Guest *resource.Quantity `json:"guest,omitempty"`
+
+	// Hugepages allow to use hugepages for the VirtualMachineInstance instead of regular memory.
+	// +optional
+	Hugepages *v1.Hugepages `json:"hugepages,omitempty"`
+}
+```
+
+###  [`VirtualMachineFlavor`](http://kubevirt.io/api-reference/main/definitions.html#_v1alpha1_virtualmachineflavor) and [`VirtualMachineClusterFlavor`](http://kubevirt.io/api-reference/main/definitions.html#_v1alpha1_virtualmachineclusterflavor)
+
+```go
+// VirtualMachineFlavor resource contains common VirtualMachine configuration
+// that can be used by multiple VirtualMachine resources.
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +genclient
+type VirtualMachineFlavor struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// VirtualMachineFlavorSpec for the flavor
+	Spec *VirtualMachineFlavorSpec `json:"spec"`
+}
+```
+
+```go
+// VirtualMachineClusterFlavor is a cluster scoped version of VirtualMachineFlavor resource.
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +genclient
+// +genclient:nonNamespaced
+type VirtualMachineClusterFlavor struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// VirtualMachineFlavorSpec for the flavor
+	Spec *VirtualMachineFlavorSpec `json:"spec"`
+}
+```
+
+###  `VirtualMachinePreferenceSpec`
+
+```go
+// VirtualMachinePreferenceSpec
+//
+// +k8s:openapi-gen=true
+type VirtualMachinePreferenceSpec struct {
+
+	//+optional
+	Clock *ClockPreferences `json:"clock,omitempty"`
+
+	//+optional
+	CPU *CPUPreferences `json:"cpu,omitempty"`
+
+	//+optional
+	Devices *DevicePreferences `json:"devices,omitempty"`
+
+	//+optional
+	Features *FeaturePreferences `json:"features,omitempty"`
+
+	//+optional
+	Firmware *FirmwarePreferences `json:"firmware,omitempty"`
+
+	//+optional
+	Machine *MachinePreferences `json:"machine,omitempty"`
+}
+```
+
+### `ClockPreferences`
+
+```go
+// ClockPreferences contains various optional defaults for Clock.
+//
+// +k8s:openapi-gen=true
+type ClockPreferences struct {
+
+	// ClockOffset allows specifying the UTC offset or the timezone of the guest clock.
+	//
+	// +optional
+	PreferredClockOffset *v1.ClockOffset `json:"preferredClockOffset,omitempty"`
+
+	// Timer specifies whih timers are attached to the vmi.
+	//
+	// +optional
+	PreferredTimer *v1.Timer `json:"preferredTimer,omitempty"`
+}
+```
+
+### `CPUPreferences`
+
+```go
+// +k8s:openapi-gen=true
+type PreferredCPUTopology string
+
+const (
```

Comment:
In the code we want to have API documentation for these consts

Reply:
ACK